src/app/ClusterInfo.h

```diff
@@ -25,17 +25,39 @@ namespace chip {
 namespace app {
 struct ClusterInfo
 {
-    ClusterInfo(const AttributePathParams & aAttributePathParams, bool aDirty) :
-        mAttributePathParams(aAttributePathParams), mDirty(aDirty)
-    {}
+    enum class Type : uint8_t
+    {
+        kInvalid        = 0,
+        kFieldIdValid   = 0x01,
+        kListIndexValid = 0x02,
+        kEventIdValid   = 0x03,
+    };
+
     ClusterInfo() {}
     bool IsDirty() { return mDirty; }
     void SetDirty() { mDirty = true; }
     void ClearDirty() { mDirty = false; }
-    bool IsSamePath(const ClusterInfo & other) const { return other.mAttributePathParams.IsSamePath(mAttributePathParams); }
-    AttributePathParams mAttributePathParams;
-    bool mDirty          = false;
-    ClusterInfo * mpNext = nullptr;
+    NodeId mNodeId         = 0;
+    ClusterId mClusterId   = 0;
+    ListIndex mListIndex   = 0;
+    FieldId mFieldId       = 0;
+    EndpointId mEndpointId = 0;
+    bool mDirty            = false;
+    Type mType             = Type::kInvalid;
+    ClusterInfo * mpNext   = nullptr;
+    EventId mEventId       = 0;
+    /* For better structure alignment
```

Comment:
This comment may not be as clear. Maybe state as:

above ordering is by bit-size to ensure least amount of memory alignment padding. Changing order to something more natural (e.g. clusterid before nodeid) will result in extra memory alignment padding.

Reply:
Thanks, done, agree.